### ISSUE TITLE
AzSK_ARMTemplateChecker - VNET failure False positive

### ISSUE THREAD

**Issue by samhodgkinson** [azsk/DevOpsKit#496]
## Title</br>

AzSK_ARMTemplateChecker - VNET failure False positive

## Description</br>

When running the AzSK_ARMTemplateChecker, from with Azure DevOps, my VNET template fails with the error Azure_VNet_NetSec_Justify_IPForwarding_for_NICs.

My template only contains 
Microsoft.Network/virtualNetworks
Microsoft.Network/networkSecurityGroups
Microsoft.Network/routeTables

There's no interface and none with IPForwarding enabled. 

## Steps to reproduce
Azure Devops CI/CD Build Pipeline with the AzSK_ARMTemplateChecker

checking template with 
Microsoft.Network/virtualNetworks
Microsoft.Network/networkSecurityGroups
Microsoft.Network/routeTables

Error
Azure_VNet_NetSec_Justify_IPForwarding_for_NICs

## Expected behavior</br> 
Only alert when an Interface is within the template with IPForwarding is enabled. 
## Actual behavior</br>

When running the AzSK_ARMTemplateChecker, from with Azure DevOps, my VNET template fails with the error Azure_VNet_NetSec_Justify_IPForwarding_for_NICs.

My template only contains 
Microsoft.Network/virtualNetworks
Microsoft.Network/networkSecurityGroups
Microsoft.Network/routeTables

There's no interface and none with IPForwarding enabled.

**Comment by Khushboo-msft** [comment]
@samhodgkinson Could you please post the detailed error that you got. _Azure_VNet_NetSec_Justify_IPForwarding_for_NICs_ is the control Id and not an error.

**Comment by samhodgkinson** [comment]
Your right; but the issue here is, IPForwarding is only enabled on the interface. I have scanned a template just containing VNET/Subnet/NSG. I should only get an IPForwarding alert/error when an interface is deployed and the enableIPForwarding parameter is set to $true. 

Interface Schema 
https://docs.microsoft.com/en-us/azure/templates/microsoft.network/2018-11-01/networkinterfaces

Network Schema
https://docs.microsoft.com/en-us/azure/templates/microsoft.network/2018-11-01/virtualnetworks

I will run the scan later and add a screen grab.

**Comment by patrick138** [comment]
same issue here:

ControlId          : Azure_VNet_NetSec_Justify_IPForwarding_for_NICs
FeatureName        : VirtualNetwork
Status             : Failed
SupportedResources : Microsoft.Network/virtualNetworks , Microsoft.Network/networkInterfaces
Severity           : High
PropertyPath       : Not found
LineNumber         : -1
CurrentValue       :
ExpectedProperty   : $.properties.enableIPForwarding
ExpectedValue      : 'False'
ResourcePath       : resources[0]
ResourceLineNumber : 86
Description        : Use of IP Forwarding on any NIC in a virtual network should be scrutinized
FilePath           : D:\a\1\s\src\Generic-VNet-and-SNet\vnet.json

**Comment by ashishrajsrivastava** [comment]
Yep getting the same false error for Vnet resource. If I add the expected property then it passes the azsk validation but then in deployment it will fail as its not a valid schema for Vnet.

`##[error]BadRequest: {
  "error": {
    "code": "InvalidRequestFormat",
    "message": "Cannot parse the request.",
    "details": [
      {
        "code": "InvalidJson",
        "message": "Could not find member 'enableIPForwarding' on object of type 'VirtualNetworkProperties'. Path 'properties.enableIPForwarding', line 1, position 115."
      }
    ]
  }
} undefined`

**Comment by Khushboo-msft** [comment]
This will be fixed in our next release (going out this week).
